src/lib/Auth.js

```diff
@@ -0,0 +1,170 @@
+   * @param dateUTC {string} like '2019-03-11T17:05:00.969'
+   * @returns {string}
+   */
+  getSignature({httpUrl, dateUTC}) {
+    const httpMethod   = 'POST';
+    const apiKeyCsod   = 'x-csod-api-key:' + this.apiId;
+    const dateCsod     = 'x-csod-date:' + dateUTC;
+    const stringToSign = httpMethod + '\n' + apiKeyCsod + '\n' + dateCsod + '\n' + httpUrl;
+    const secretKey    = new Buffer(this.apiSecret, 'base64');
+    const hmac         = crypto.createHmac('sha512', secretKey);
+    console.log('[getSignature] - signature: ', JSON.stringify(stringToSign));
+
+    return hmac.update(stringToSign).digest('base64');
+  }
+
+  getSignatureSession({method, sessionToken, sessionSecret, httpUrl, dateUTC}) {
+    const sessionTokenKey = 'x-csod-session-token:' + sessionToken;
+    const dateCsod        = 'x-csod-date:' + dateUTC;
+    const stringToSign    = method + '\n' + dateCsod + '\n' + sessionTokenKey + '\n' + httpUrl;
+    const secretKey       = new Buffer(sessionSecret, 'base64');
+    const hmac            = crypto.createHmac('sha512', secretKey);
+    console.log('[getSignatureSession] - signature: ', JSON.stringify(stringToSign));
+
+    return hmac.update(stringToSign).digest('base64');
+  }
+
+  /**
+   * Connection for beginning request
+   * @param baseUrl
+   * @param apiKey
+   * @param dateUTC
+   * @param signature
+   * @returns {Promise<*>}
+   */
+  async setConnection({baseUrl, apiKey, dateUTC, signature}) {
+
+    return await axios.create({
+      baseUrl: baseUrl,
+      timeout: 50000,
+      headers: {
+        'x-csod-api-key':   apiKey,
+        'x-csod-date':      dateUTC,
+        'x-csod-signature': signature
+      }
+    });
+  }
+
+  async setConnectionSession({baseUrl, dateUTC, token, signature}) {
+
+    return await axios.create({
+      baseUrl: baseUrl,
+      timeout: 50000,
+      headers: {
+        'x-csod-date':          dateUTC,
+        'x-csod-session-token': token,
+        'x-csod-signature':     signature
+      }
+    });
+  }
+
+  /**
+   * Get session from cornerstone
+   * @returns {Promise<{alias: *, expiresOn: *, secret: *, status: number | string, token: *}>}
+   */
+  async setSession({dateUTC}) {
+    const sessionFile = await JSON.parse(this.readSession());
+    if (sessionFile) {
+      const dateNow     = new Date();
+      const dateSession = new Date(sessionFile.expiresOn);
+
+      if (dateNow < dateSession) {
+        return sessionFile;
+      }
+    }
+
+    const httpUrl  = config.CORNERSTONE_PATH_SESSION;
+
+    const signature = this.getSignature({
+      apiId:     this.apiId,
+      apiSecret: this.apiSecret,
+      httpUrl:   httpUrl,
+      dateUTC:   dateUTC
+    });
+
+    const baseUrl = this.getBaseUrl({corpname: this.corpname});
+
+    const path = baseUrl + httpUrl + '?userName={username}&alias={alias}'
+      .replace('{username}', this.username)
+      .replace('{alias}', this.alias + Date.now());
```

Comment:
You shoud use template strings for this!
```js
const path = `${baseUrl}${httpUrl}?userName=${username}&alias=${this.alias + Date.now()}';
```

Reply:
You're right, it's more eleguant. It's done